packages/dom/src/dom.js

```diff
@@ -655,3 +655,54 @@ export function wrap( newNode, referenceNode ) {
 	referenceNode.parentNode.insertBefore( newNode, referenceNode );
 	newNode.appendChild( referenceNode );
 }
+
+function maybeUnwrapEvent( event ) {
+	// In some contexts, it may be necessary to capture and redirect the
+	// event (e.g. atop an `iframe`). To accommodate this, you can
+	// create an instance of CustomEvent with the original event specified
+	// as the `detail` property.
+	//
+	// See: https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Creating_and_triggering_events
+	return window.CustomEvent && event instanceof window.CustomEvent ? event.detail : event;
+}
+
+/**
+ * Whether the event coordinates are within the element bounds.
+ *
+ * @since 2.1.0
+ *
+ * @param {Element} element The element to check.
+ * @param {Event} event The event that tracks the coordinates.
+ * @return {boolean} True if is within bounds, false if not.
+ */
+export function isWithinBounds( element, event ) {
+	const { clientX: x, clientY: y } = maybeUnwrapEvent( event );
+
+	const rect = element.getBoundingClientRect();
+	/// make sure the rect is a valid rect
+	if ( rect.bottom === rect.top || rect.left === rect.right ) {
+		return false;
+	}
+
+	return x >= rect.left && x <= rect.right && y >= rect.top && y <= rect.bottom;
+}
+
+/**
+ * Calculates which axis the event is closer to (`left`/`right`, `top`/`bottom`).
```

Comment:
I'd have expected a function by this name to return some set of coordinates. The `left`, `right`, `top`, `bottom` thing seems like a pretty implementation-specific detail. Maybe at the least I'd call it something like `getClosestEdge`. Is there a reason for making this reusable?

Reply:
I like that name better! Fixed at ac6fbd0 

As for the reason to make this reusable: I think is worth abstracting away the complexity of these operations from consumers.